README.md

```diff
@@ -1,14 +1,62 @@
+
+A special rule called the "flying general" applies to the general, where if the two generals from opposing side faces each other in an open (no pieces blocking) vertical column, the general can "fly" and capture the opponent's general. Since a general will occupy an open vertical column first, the opponent will not be able to move their general into the open column since that will be a movement into check. However, this rule can be used to checkmate an opponent's general.
+
+### The Advisor
+Similar to the general, the advisor can only move within the palace. In contrast to the general, the advisor can only move diagonally, which means that there are only 5 possible positions that each advisor can take. The advisor can be used to capture a piece as it moves.
+
+### The Elephant
+The elephant moves exactly 2 steps diagonally, which can also be visualized as two positions to the left/right and two positions forward/backward. The elephant will not be able to jump over pieces while moving, meaning if a piece exists to the top right of the elephant, it will not be able to jump over that piece to reach the top right position of the piece.
+
+The elephant can be used to a piece as it moves. The elephant cannot cross the river, which is why it is mainly used as a defensive piece.
+
+### The Horse
+The horse in Xiang Qi behaves similarly to knight in chess, where it moves following the "L" shape, which can be visualized as moving 1 position orthogonally followed by 1 position diagonally away from the horse. The horse can be used to capture a piece as it moves.
+
+In addition to the rules in chess, the horse in Xiang Qi can be blocked when a piece is placed orthogonally to the horse. The moves that are blocked by a specific piece can be visualized as whichever moves which will touch a piece when making the "first" orthogonal move. For example, when a piece is placed in the forward position of a horse, the horse will not be able to make the following 2 moves: horse lands at 2 position forward and 1 position to the right, and 2 position forward and 1 position to the left. However, other moves will still be available for the horse.
+
+### The Chariot
+The chariot in Xiang Qi behaves exactly like rook in chess, where it can move orthogonally for however many positions without skipping over a piece. The chariot can be used to capture a piece as it moves.
+
+### The Cannon
+The cannon is the only piece in the game where its movement and capturing pattern is different. The movement of the cannon is similar to the chariot, where it can move orthogonally for however many positions without skipping over a piece.
+
+Unlike the chariot, the cannon must capture a piece by skipping over another piece. There can be be any number of blank positions that exists between the cannon to the piece that it is skipping over, and from the piece that it is skipping over to the piece being captures.
```

Comment:
I think it would be nice to clarify that cannons can skip both ally and enemy pieces.

Reply:
The very last word seems to be changed: captures -> captured

Reply:
Done